app/scripts/controllers/edit/deploymentConfig.js

```diff
@@ -0,0 +1,289 @@
+              resolve: {
+                modalConfig: function() {
+                  return {
+                    alerts: $scope.alerts,
+                    message: "Some of your existing configuration can be reused in the picked " + pickedStrategy + " strategy configuration. Do you want reuse it?",
+                    details: "Reusing configuration will remove " + $scope.originalStrategy + " strategy configuration after the changes are saved.",
+                    okButtonText: "Yes",
+                    okButtonClass: "btn-primary",
+                    cancelButtonText: "No"
+                  };
+                }
+              }
+            });
+
+            modalInstance.result.then(function () {
+              // Move parameters that belong to the origial strategy to the picked one.
+              $scope.strategyData[pickedStrategyParams] = $scope.strategyData[getParamsString($scope.originalStrategy)];
+              $scope.paramsMoved = getParamsString($scope.originalStrategy);
+            }, function() {
+              // Create empty parameters for the newly picked strategy
+              $scope.strategyData[pickedStrategyParams] = {};
+            });
+          } else {
+          }
+          break;
+        default:
+          if (!_.has($scope.strategyData, pickedStrategyParams)) {
+            if (pickedStrategy !== 'Custom') {
+              $scope.strategyData[pickedStrategyParams] = {};
+            } else {
+              $scope.strategyData[pickedStrategyParams] = {
+                image: "",
+                command: [],
+                environment: []
+              }
+            } 
+          }
+          break;
+      }
+      $scope.displayedParams = getParamsString(pickedStrategy);
+    };
+
+    var getParamsString = function(strategyType) {
+      return strategyType.toLowerCase() + 'Params';
+    }
+
+    // helper for detemining if strategy switch was done between Rolling <-> Recreate strategy
+    var isRollingRecreateSwitch = function(pickedStrategy) {
+      return (pickedStrategy !== 'Custom' && $scope.originalStrategy !== 'Custom' && pickedStrategy !== $scope.originalStrategy);
+    };
+
+    var assembleImageChangeTrigger = function(containerName, ist, trigger) {
+      var istagObject = {
+        kind: "ImageStreamTag",
+        namespace: ist.namespace,
+        name: ist.imageStream + ':' + ist.tagObject.tag
+      }
+      if (trigger) {
+        trigger.imageChangeParams.from = istagObject;
+      } else {
+        trigger = {
+          type: "ImageChange",
+          imageChangeParams: {
+            automatic: true,
+            containerNames: [containerName],
+            from: istagObject
+          }
+        };
+      }
+      return trigger;
+    };
+
+    var updateTriggers = function() {
+      var updatedTriggers = _.remove($scope.updatedDeploymentConfig.spec.triggers, function(trigger) {return trigger.type !== "ImageChange"});
```

Comment:
It's kinds weird since but in fact it's vice versa... not sure if they have mistake in their docs but this will return array of non-ICT triggers